### PR TITLE
csgrep --tool: filter defects by the tool that detected them

### PR DESCRIPTION
**Pull request by kdudka**

### CODE REVIEW

File: src/csgrep.cc [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include "cswriter.hh"
 #include "json-writer.hh"
 #include "regex.hh"
+#include "parser-common.hh"
```
Comment (lzaoral) [inline src/csgrep.cc:26]: Sort these alphabetically, please.

File: src/csgrep.cc [hunk 2]
```diff
@@ -194,6 +195,26 @@ class MsgPredicate: public IPredicate {
...
+        const ImpliedAttrDigger digger_;
+        const RE re_;
+
+    public:
+        ToolPredicate(const RE &re):
+            re_(re)
+        {
+        }
+
+        virtual bool matchDef(const Defect &defOrig) const {
```
Comment (lzaoral) [inline src/csgrep.cc:209]: Replace `virtual` with `override` please.

File: src/csgrep.cc [hunk 3]
```diff
@@ -499,6 +520,7 @@ bool chainFiltersCore(
         && appendPredIfNeeded<ErrorPredicate>     (pf, vm, flags, "error")
         && appendPredIfNeeded<KeyEventPredicate>  (pf, vm, flags, "event")
         && appendPredIfNeeded<MsgPredicate>       (pf, vm, flags, "msg")
+        && appendPredIfNeeded<ToolPredicate>      (pf, vm, flags, "tool")
```
Comment (lzaoral) [inline src/csgrep.cc:523]: Sort these alphabetically, please.

File: src/csgrep.cc (hunk 1 above)
Comment (kdudka) [reply src/csgrep.cc]: Oops, I thought I did.

File: src/csgrep.cc (hunk 3 above)
Comment (kdudka) [reply src/csgrep.cc]: Fixed with commit 5a35bd926806bdde3964122b56547a4b2cf39cfe.

**Review by lzaoral — APPROVED** [review]

**Pull request merged**